components/Composition/FAnnouncementCard.jsx

```diff
@@ -0,0 +1,88 @@
+import { StyleSheet, View } from 'react-native';
+import { FImage } from 'components/Composition/FImage';
+import { getHalfBorderRadius } from 'utils/getHalfBorderRadius';
+import colors from 'themes/colors';
+import sizes from 'themes/sizes';
+import opacities from 'themes/opacities';
+import fonts from 'themes/fonts';
+import { FHeading } from 'components/Composition/FHeading';
+import React from 'react';
+import { TouchableWithoutFeedback } from 'react-native-web';
+import { FHeadingWithIcon } from 'components/Composition/FHeadingWithIcon';
+import placements from 'themes/placements';
+import icons from 'themes/icons';
+
+export const FAnnouncementCard = ({ width, data }) => {
+  const cropText = (text, number) => `${text.slice(0, width / number)}...`;
```

Comment:
takie rzeczy da sie osiągnąc bez funkcji.
https://stackoverflow.com/questions/30594080/how-to-have-ellipsis-effect-on-text
Trzeba tylko dorobić propsy do FHeading, tylko tak żeby nie zepsuć
Nowe propsy
1, numberOfLines to Text jako atrybut
2. ellipsizeMode
Żeby Ci ucinało dajesz numberOfLines  = 1 a mode na 'clip'

To są defaultowe propsy wiec jak ich gdzieś nie podasz to nie powinny mieć żadnego efektu na widok. Przetestuj kilka widoków